client/src/components/login-signup/Signup.js

```diff
@@ -1,15 +1,38 @@
-import React from "react";
+import React, { useState, useRef, useEffect } from "react";
+import AuthService from '../../Services/AuthService'
 import { useForm } from "react-hook-form";
+import Message from '../Message/Message';
 import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
 
-export default function Register() {
-  const { register, handleSubmit } = useForm();
+export default function Register(props) {
+  const { register, handleSubmit, reset } = useForm();
 
-  const onSubmit = (data) => {
-    console.log(data);
+  const [message, setMessage] = useState(null);
+  let timerID = useRef(null);
+
+  useEffect(() => {
+    return () => {
+      clearTimeout(timerID);
+    };
+  }, []);
+
+  const onSubmit = (user,e) => {
+    AuthService.register(user).then((data) => {
+      const { message } = data;
+      setMessage(message);
+      e.target.reset();
```

Comment:
What does this reset do? Comments would be helpful

Reply:
it's coming from hooks, userForm() method. after we got data from the server, we clearing the user input data in Signup page, I will added all the comments with next pull request.